src/Pos/Util/UserSecret.hs

```diff
@@ -81,19 +96,50 @@ instance Bi UserSecret where
         keys <- get
         return $ def & usVss .~ vss & usPrimKey .~ pkey & usKeys .~ keys
 
+#ifdef POSIX
+-- | Constant that defines file mode 600.
+mode600 :: PSX.FileMode
+mode600 = PSX.unionFileModes PSX.ownerReadMode PSX.ownerWriteMode
+
+-- | Check whether a given file has mode 600.
+failIfModeNot600 :: (MonadIO m, MonadFail m) => FilePath -> m ()
+failIfModeNot600 path = do
+    mode <- liftIO $ PSX.fileMode <$> PSX.getFileStatus path
+    when (mode /= mode600) $
+        fail "Secret file mode incorrect. Set it to 600 and try again."
+
+-- | Set mode 600 on a given file, regardless of its current mode.
+setMode600 :: (MonadIO m) => FilePath -> m ()
+setMode600 path = liftIO $ PSX.setFileMode path mode600
+#endif
+
 -- | Create user secret file at the given path, but only when one doesn't
 -- already exist.
 initializeUserSecret :: (MonadIO m) => FilePath -> m ()
 initializeUserSecret path = do
     exists <- T.testfile (fromString path)
-    liftIO (if exists
-            then return ()
-            else T.output (fromString path) empty)
+    liftIO $
+#ifdef POSIX
+        if exists
+        then failIfModeNot600 path
```

Comment:
@startling do you think it is ok to fail? This is our concern. For example user wants to run cardano-node with a key which is on disk, but it has wrong permissions. We can:
 * fail and notify user that he should manually change permissions
 * using steps described above (https://github.com/input-output-hk/cardano-sl/pull/98#discussion_r105655784) just silently fix permissions

SSH fails in this situation. Should we do the same?